testcases/parser/nodes.go

```diff
@@ -43,6 +45,77 @@ type TestCase struct {
 	FuncType      TestFuncType
 }
 
+func (tc *TestCase) GetFunctionOptions() []*types.FunctionOption {
+	if len(tc.Options) == 0 {
+		return nil
+	}
+	funcOptions := make([]*types.FunctionOption, 0)
+	for key, value := range tc.Options {
+		funcOptions = append(funcOptions, &types.FunctionOption{
+			Name:       key,
+			Preference: []string{value},
+		})
+	}
+	return funcOptions
+}
+
+func (tc *TestCase) scalarSignatureKey() string {
+	var b strings.Builder
+	for i, a := range tc.Args {
+		if i != 0 {
+			b.WriteByte('_')
+		}
+		b.WriteString(a.Type.ShortString())
+	}
+	return b.String()
+}
+
+func (tc *TestCase) aggregateSignatureKey() string {
+	var b strings.Builder
+	for i, a := range tc.AggregateArgs {
+		if i != 0 {
+			b.WriteByte('_')
+		}
+		b.WriteString(a.ColumnType.ShortString())
+	}
+	return b.String()
+}
+
+func (tc *TestCase) signatureKey() string {
+	if tc.FuncType == ScalarFuncType {
```

Comment:
Consider using a switch statement here so that when support for window is added it is handled instead of returning aggregate.